### PR TITLE
Added Dagger in matrix.h

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Added in daggers. Would be a very useful feature downstream for TFQ to have.

### CODE REVIEW

File: lib/matrix.h [hunk 1]
```diff
@@ -75,6 +75,15 @@ inline void Matrix2Multiply(const Array1& u, Array2& matrix) {
   }
 }
 
+// Dagger a 2x2 matrix.
+template <typename Array>
+inline void Matrix2Dagger(Array& matrix) {
+  for(int i = 1;i < 8;i+=2) {
```
Comment (sergeisakov) [inline lib/matrix.h:81]: Fix spaces and use unsigned for consistency: `for (unsigned i = 1; i < 8; i += 2) {`.

File: lib/matrix.h [hunk 2]
```diff
@@ -198,6 +207,20 @@ inline void Matrix4Permute(Array2& matrix) {
   std::swap(matrix[26], matrix[28]); std::swap(matrix[27], matrix[29]);
 }
 
+// Dagger a 4x4 matrix.
+template <typename Array2>
+inline void Matrix4Dagger(Array2& matrix) {
+  for(int i = 1;i < 32;i+=2) {
```
Comment (sergeisakov) [inline lib/matrix.h:213]: Fix spaces and use unsigned for consistency: `for (unsigned i = 1; i < 32; i += 2) {`.

File: lib/matrix.h [hunk 3]
```diff
@@ -75,6 +75,15 @@ inline void Matrix2Multiply(const Array1& u, Array2& matrix) {
   }
 }
 
+// Dagger a 2x2 matrix.
+template <typename Array>
```
Comment (sergeisakov) [inline lib/matrix.h:79]: Replace `Array` with `Array2` for consistency.

**Comment by MichaelBroughton** [comment]
Added the two spaces :)

**Pull request merged**